src/execution_impl.rs

```diff
@@ -13,88 +16,119 @@ use wasmer::{
 use wasmer_as::{Read as ASRead, StringPtr, Write as ASWrite};
 use wasmer_middlewares::Metering;
 
-/// `Print` ABI called by the webassembly VM
-fn print(env: &Env, arg: i32) {
-    let str_ptr = StringPtr::new(arg as u32);
-    println!(
-        "{}",
-        str_ptr
-            .read(env.wasm_env.memory.get_ref().expect("initialized memory"))
-            .unwrap()
-    );
+/// Utility function to call a WASM module (compiled from AssemblyScript) using
+/// high-level Rust types (by de/serializing a String).
+///
+/// The intend is to be able to expose any kind of function as an ABI:
+///
+/// ```ignore
+/// fn get_status() -> Result<NodeStatus> {
+///     Ok(typed_call(env, "0xAddressOfSCInLedger", "get_status", ())?)
+/// }
+/// ```
+pub fn typed_call<T: Serialize, R: DeserializeOwned>(
+    env: &Env,
+    address: &Address,
+    function: &str,
+    _args: T, // TODO: @adrien-zinger how do we pass function args in AssemblyScript?
```

Comment:
@adrien-zinger

Reply:
:thinking: this is work in progress by @yvan-sraka, as I see for now this function isn't included in the instance's ImportObject so never used. Or used? Yvan?

Reply:
The purpose was to write a high-level Rust wrapper of `call_module` (which is the one pass to `ImportObject`), exposing beautiful types and hiding the magic of de/serialization on us to be able to change it in the future!

Reply:
We can still pass args "the high level way" https://github.com/massalabs/sc-runtime/discussions/18 :wink:

https://github.com/massalabs/sc-runtime/blob/wasi-env/examples/hello.js

https://github.com/massalabs/sc-runtime/blob/8f2c3fe3a5e42342d27754ca95cf4f5b1902afea/src/main.rs#L45-L57

Reply:
@AureliaDolo you write a solution here https://github.com/massalabs/sc-runtime/pull/14 ?!

Reply:
yep